plugins/groovy/groovy-psi/src/org/jetbrains/plugins/groovy/codeInspection/changeToOperator/data/MethodCallData.java

```diff
@@ -0,0 +1,138 @@
+
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.tree.IElementType;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.plugins.groovy.lang.psi.GroovyPsiElementFactory;
+import org.jetbrains.plugins.groovy.lang.psi.api.statements.expressions.*;
+import org.jetbrains.plugins.groovy.lang.psi.impl.utils.ParenthesesUtils;
+
+import static org.jetbrains.plugins.groovy.lang.psi.impl.utils.BoolUtils.isNegation;
+import static org.jetbrains.plugins.groovy.lang.psi.impl.utils.ComparisonUtils.isComparison;
+
+public final class MethodCallData {
+  private final PsiElement backingElement;
+
+  private boolean isNegated;
+  private GrExpression base;
+  private GrExpression[] arguments;
+  private IElementType comparison;
+
+  @Nullable
+  public static MethodCallData create(@NotNull PsiElement backingElement) {
+    return Builder.build(backingElement);
+  }
+
+  private MethodCallData(@NotNull PsiElement backingElement) {
+    this.backingElement = backingElement;
+  }
+
+  @NotNull
+  public PsiElement getBackingElement() {
+    return backingElement;
+  }
+
+  public boolean isNegated() {
+    return isNegated;
+  }
+
+  @NotNull
+  public String getBase() {
+    return base.getText();
+  }
+
+  @NotNull
+  public String getArgument(int i) {
+    return arguments[i].getText();
+  }
+
+  @Nullable
+  public IElementType getComparison() {
+    return comparison;
+  }
+
+  private static final class Builder {
+    @SuppressWarnings("UnusedAssignment")
+    @Nullable
+    public static MethodCallData build(@NotNull PsiElement backingElement) {
+      MethodCallData result = new MethodCallData(backingElement);
+
+      PsiElement element = backingElement;
+      element = handleNegation(result, element);
+      element = handleComparison(result, element);
+      element = handleMethodCall(result, element);
+      return result;
+    }
+
+    private static PsiElement handleNegation(MethodCallData result, PsiElement element) {
+      result.isNegated = isNegation(element);
+
+      if (result.isNegated) element = ((GrUnaryExpression)element).getOperand();
+
+      return element;
+    }
+
+    private static PsiElement handleComparison(MethodCallData result, PsiElement element) {
+      if (isComparison(element)) {
+        GrBinaryExpression relationalExpression = (GrBinaryExpression)element;
+
+        element = relationalExpression.getLeftOperand();
+        result.comparison = relationalExpression.getOperationTokenType();
+      }
+      return element;
+    }
+
+    @Nullable
+    private static PsiElement handleMethodCall(MethodCallData result, PsiElement element) {
+      if (!(element instanceof GrMethodCall)) return null;
+
+      GrMethodCall methodCall = (GrMethodCall)element;
+      GrReferenceExpression invokedExpression = (GrReferenceExpression)methodCall.getInvokedExpression();
+
+      result.base = addParenthesesIfNeeded(invokedExpression.getQualifierExpression());
```

Comment:
What if qualifier expression is null?
for example `bar(foo)` is method call, invoked expression is `bar` and it has no qualifier.